tests/integration/backend.go

```diff
@@ -0,0 +1,60 @@
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethclient/simulated"
+	"github.com/stretchr/testify/require"
+)
+
+type Backend interface {
```

Comment:
this interface provides a small abstraction so that we may use geth's in-memory simulated backend or the rpc client (connected to the testcontainer running h geth in dev mode) interchangeably (and without clunky casts).